lib/notifications/display.dart

```diff
@@ -118,10 +109,15 @@ class NotificationDisplayManager {
 
   static Future<void> _onMessageFcmMessage(MessageFcmMessage data, Map<String, dynamic> dataJson) async {
     assert(debugLog('notif message content: ${data.content}'));
+
+    final globalStore = await ZulipBinding.instance.getGlobalStore();
     final zulipLocalizations = GlobalLocalizations.zulipLocalizations;
     final groupKey = _groupKey(data);
     final conversationKey = _conversationKey(data, groupKey);
 
+    final intentUri = await notificationIntentUriFromFcmMessage(globalStore: globalStore, data: data);
```

Comment:
This has a similar latency issue to the one in my previous comment on opening the notification.

Here, I think the symptom would be: if a notification arrives and the app isn't already running (or doesn't yet have data for the account the notification is for), we'd end up having to sit here for however long it takes to load the per-account data. That could be a while — even a few seconds (like users of chat.zulip.org would see when on a good network connection on the same continent as the server) is enough to make notifications feel laggy in the context of a messaging application.

For users on a flaky or slow network connection, especially in a big realm like chat.zulip.org, the delay could be more like a minute, and could be enough to make the OS time out and stop the app, so the notification doesn't appear at all. In particular I think if the device is asleep it might (though I'm not sure) remain in a lower-power state even while handling a notification, which would further slow down the download of a couple of MB of server data.

Unlike the situation when opening a notification, this would affect even users who have only one account.

So I think we definitely want to avoid pulling in the per-account store on the path to displaying a notification. We can do that with the same solution discussed at the other end: just encode the notification's narrow information directly, rather than converting to a full Zulip narrow URL.

I guess this means the `getGlobalStore` method from #1005 probably isn't needed right now after all; oh well.

At some point in the future (solidly post-launch), we'll want to use per-account server data in displaying a notification, to do things like showing the names of other recipients in a DM thread. A prerequisite for that will be to store server data in the app's local database, so that we can load a version for it without waiting to load everything from the server from scratch.